In [6]:
#!pip install ast

In [3]:
import pandas as pd
import numpy as np
import ast

/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)


In [4]:
content_df = pd.read_csv("/Users/shriyashekhar/Documents/IWork/Data/Output/content_nouns.csv", encoding="ISO-8859-1")
content_df.head()

StreamID                                            Content  \
0       163  TXmAk2KZAy4\r\nNMeUjebo1Ac\r\nEEuTxFhp3go\r\nC...   
1       419  TXmAk2KZAy4\r\nNMeUjebo1Ac\r\nEEuTxFhp3go\r\nC...   
2       507  wBYKUgUyGWc\r\nA team of world-class drivers, ...   
3       199  Castrol EDGE  is Castrol?s flagship power bran...   
4       201  Charles ?Cheers? Wakefield, Castrol?s founder,...   

                                       Content_Nouns  
0  [('Castrol', 'NNP'), ('oil', 'NN'), ('TITANIUM...  
1  [('Castrol', 'NNP'), ('oil', 'NN'), ('TITANIUM...  
2  [('wBYKUgUyGWc', 'NN'), ('team', 'NN'), ('worl...  
3  [('Castrol', 'NNP'), ('EDGE', 'NNP'), ('Castro...  
4  [('Charles', 'NNP'), ('\x93Cheers\x94', 'NNP')...

In [5]:
def get_nouns_for_stream(stream_id):
    stream_nouns = content_df[content_df["StreamID"] == stream_id]["Content_Nouns"].values
    #stream_nouns = stream_nouns.astype(str)
    first_stream_noun = stream_nouns[0]
#     print("Printing debug info....")
#     print(first_stream_noun)
#     print(type(first_stream_noun))
#     print("End of printing debug info....")
    
    if isinstance(first_stream_noun, str):
        # https://stackoverflow.com/questions/10775894/converting-a-string-representation-of-a-list-into-an-actual-list-object
        first_stream_noun = ast.literal_eval(first_stream_noun)
        
    return [x[0] for x in first_stream_noun]

### Nouns for the stream 507 as a list as below...

In [6]:
get_nouns_for_stream(199)

['Castrol',
 'EDGE',
 'Castrol\x92s',
 'flagship',
 'power',
 'brand',
 'PCOs',
 'performance',
 'engine',
 'oil',
 'performing',
 'vehicles',
 'OEMs',
 'world',
 'Castrol',
 'EDGE',
 'TITANIUM',
 'FST\x99',
 'oil',
 'strength',
 'oil',
 'film',
 'protection',
 'oil',
 'pressures',
 'engines',
 'today',
 'video',
 'Today',
 'engines',
 'pressure',
 'areas',
 'engine',
 'pressures',
 'excess',
 'square',
 'centimetre',
 'that\x92s',
 'cars',
 'stiletto',
 'hee',
 'video',
 'Castrol',
 'EDGE',
 'strength',
 'technology',
 'pressures',
 'high-performance',
 'engines',
 'Castrol',
 'EDGE',
 'problem',
 'engine',
 'pressures',
 'TITANIUM',
 'FST\x99',
 'Fluid',
 'Strength',
 'Technology',
 'TITANIUM',
 'FST\x99',
 'oil',
 'strength',
 'oil',
 'breakdown',
 'sha',
 'TITANIUM',
 'FST\x99',
 'adapts',
 'engine',
 'need',
 'pressures',
 'contact',
 'variety',
 'engine',
 'spee',
 'Castrol',
 'R',
 'amp',
 'D',
 'Technology',
 'world',
 'Castrol',
 'Stress',
 'Test',
 'film',
 'strength',
 'Cast

## Shriya: So you call the method like this: get_nouns_for_stream(507) to get a list of nouns for that stream with the specified stream ID. You can then loop through the items in the list and do your word net and SPARQL analysis.

In [7]:
from nltk.corpus import wordnet as wn

list_of_nouns = get_nouns_for_stream(201)
print(len(list_of_nouns))

for i in range(0, len(list_of_nouns)):
    for j in range(0, len(list_of_nouns)):
# for noun in list_of_nouns:
#     print(i)
#     if not wn.synsets(list_of_nouns[i]):
        while(len(wn.synsets(list_of_nouns[i]))==0):
            i+=1
        a = wn.synsets(list_of_nouns[i])[0]
#         i+=1
        while(len(wn.synsets(list_of_nouns[j]))==0):
    #         print(i)
            j+=1
        print(j)
        b = wn.synsets(list_of_nouns[j])[0]
        if a and b:
            print('two terms')
            print(a)
            print(b)
            print('parent')
            print(a.lowest_common_hypernyms(b))


# return a.lowest_common_hypernyms(b)

100
0
two terms
Synset('charles.n.01')
Synset('charles.n.01')
parent
[Synset('charles.n.01')]
4
two terms
Synset('charles.n.01')
Synset('laminitis.n.01')
parent
[Synset('entity.n.01')]
4
two terms
Synset('charles.n.01')
Synset('laminitis.n.01')
parent
[Synset('entity.n.01')]
4
two terms
Synset('charles.n.01')
Synset('laminitis.n.01')
parent
[Synset('entity.n.01')]
4
two terms
Synset('charles.n.01')
Synset('laminitis.n.01')
parent
[Synset('entity.n.01')]
5
two terms
Synset('charles.n.01')
Synset('entrepreneur.n.01')
parent
[Synset('person.n.01')]
6
two terms
Synset('charles.n.01')
Synset('vision.n.01')
parent
[Synset('entity.n.01')]
7
two terms
Synset('charles.n.01')
Synset('lubricant.n.01')
parent
[Synset('physical_entity.n.01')]
8
two terms
Synset('charles.n.01')
Synset('business.n.01')
parent
[Synset('entity.n.01')]
10
two terms
Synset('charles.n.01')
Synset('london.n.01')
parent
[Synset('object.n.01')]
10
two terms
Synset('charles.n.01')
Synset('london.n.01')
parent
[Synset('object.

IndexError: list index out of range

In [8]:
from nltk.corpus import wordnet as wn
words = get_nouns_for_stream(201)
syns = {w : [] for w in words}

for k, v in syns.items():
    for synset in wn.synsets(k):
        for lemma in synset.lemmas():
            v.append(lemma.name())
print(syns)

{'Charles': ['Charles', 'Charles_IX', 'Charles', 'Charles_VII', 'Charles', 'Charles_II', 'Charles_I', 'Charles_the_Bald', 'Charles', 'Charles_II', 'Charles', 'Charles_I', 'Charles_Stuart', 'Charles', 'Prince_Charles', 'Charles', 'Jacques_Charles', 'Jacques_Alexandre_Cesar_Charles', 'Charlemagne', 'Carolus', 'Charles', 'Charles_I', 'Charles_the_Great', 'Charles', 'Charles_River'], '\x93Cheers\x94': [], 'Wakefield': [], 'Castrol\x92s': [], 'founder': ['laminitis', 'founder', 'founder', 'beginner', 'founding_father', 'father', 'founder', 'fall_through', 'fall_flat', 'founder', 'flop', 'founder', 'collapse', 'fall_in', 'cave_in', 'give', 'give_way', 'break', 'founder', 'founder'], 'entrepreneur': ['entrepreneur', 'enterpriser'], 'vision': ['vision', 'sight', 'vision', 'visual_sense', 'visual_modality', 'vision', 'visual_sensation', 'imagination', 'imaginativeness', 'vision', 'vision'], 'lubricant': ['lubricant', 'lubricator', 'lubricating_substance', 'lube'], 'business': ['business', 'conc

In [9]:
dog = wn.synsets('dog')[0]
cat = wn.synsets('cat')[0]
cat.lowest_common_hypernyms(dog)

[Synset('carnivore.n.01')]

In [10]:
oil = wn.synsets('oil')[1]
castrol = wn.synsets('liquid')[0]
oil.lowest_common_hypernyms(castrol)

[Synset('substance.n.01')]

In [11]:
wn.synset('policeman.n.01').lowest_common_hypernyms(wn.synset('chef.n.01'))

[Synset('person.n.01')]

In [12]:
bata = wn.synsets('Wakefield')
if bata:
    print("empty")

In [13]:
len(wn.synsets('liquid'))

11

In [14]:
wn.synsets('liquid')[0]

Synset('liquid.n.01')

In [15]:
import nltk
from nltk.corpus import wordnet as wn
dog = wn.synset('charles.n.01')
cat = wn.synset('power.n.01')
float(dog.path_similarity(cat))

0.07692307692307693

In [16]:
from nltk.corpus import wordnet as wn

list_of_nouns_1 = get_nouns_for_stream(201)
list_of_nouns_2 = get_nouns_for_stream(199)
print(len(list_of_nouns_1))

for i in range(0, len(list_of_nouns_1)-1):
    for j in range(0, len(list_of_nouns_2)-1):
# for noun in list_of_nouns:
#     print(i)
#     if not wn.synsets(list_of_nouns[i]):
        while(len(wn.synsets(list_of_nouns_1[i]))==0):
            i+=1
        a = wn.synsets(list_of_nouns_1[i])[0]
#         i+=1
        while(len(wn.synsets(list_of_nouns_2[j]))==0):
    #         print(i)
            j+=1
        print(j)
        b = wn.synsets(list_of_nouns_2[j])[0]
        if a and b:
            print('two terms')
#             print(a)
#             print(b)
            print('parent')
            print(a.lowest_common_hypernyms(b))
            x = a.path_similarity(b)
            if((x is not None) and (x >= 0.8) ):
                print('similar terms')
                print(a)
                print(b)


# return a.lowest_common_hypernyms(b)

100
1
two terms
parent
[Synset('object.n.01')]
1
two terms
parent
[Synset('object.n.01')]
3
two terms
parent
[Synset('whole.n.02')]
3
two terms
parent
[Synset('whole.n.02')]
4
two terms
parent
[Synset('entity.n.01')]
5
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
8
two terms
parent
[Synset('whole.n.02')]
9
two terms
parent
[Synset('physical_entity.n.01')]
10
two terms
parent
[Synset('entity.n.01')]
11
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
15
two terms
parent
[Synset('object.n.01')]
15
two terms
parent
[Synset('object.n.01')]
16
two terms
parent
[Synset('physical_entity.n.01')]
18
two terms
parent
[Synset('physical_entity.n.01')]
18
two terms
parent
[Synset('physical_entity.n.01')]
19
two terms
parent
[Synset('entity.n.01')]
20
two terms
parent
[Synset('physical_entity.n.01')]
21
two terms
parent
[Synset('whole.n.02')]


38
two terms
parent
[Synset('entity.n.01')]
39
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('abstraction.n.06')]
41
two terms
parent
[Synset('abstraction.n.06')]
43
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('entity.n.01')]
44
two terms
parent
[Synset('abstraction.n.06')]
45
two terms
parent
[Synset('psychological_feature.n.01')]
46
two terms
parent
[Synset('entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
51
two terms
parent
[Synset('abstraction.n.06')]
52
two terms
parent
[Synset('entity.n.01')]
53
two terms
parent
[Synset('entity.n.01')]
54
two terms
parent
[Synset('abstraction.n.06')]
56
two terms
parent
[Synset('abstraction.n.06')]
56
two terms
parent
[Synset('abstraction.n.06')]
57
two terms
parent
[Synset('abstraction.n.06')]
58
two terms
parent
[Synset('psychological_feature.n.01')]
59
two terms
parent
[Synset

11
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
15
two terms
parent
[Synset('entity.n.01')]
15
two terms
parent
[Synset('entity.n.01')]
16
two terms
parent
[Synset('abstraction.n.06')]
18
two terms
parent
[Synset('abstraction.n.06')]
18
two terms
parent
[Synset('abstraction.n.06')]
19
two terms
parent
[Synset('abstraction.n.06')]
20
two terms
parent
[Synset('abstraction.n.06')]
21
two terms
parent
[Synset('psychological_feature.n.01')]
22
two terms
parent
[Synset('psychological_feature.n.01')]
23
two terms
parent
[Synset('abstraction.n.06')]
24
two terms
parent
[Synset('entity.n.01')]
25
two terms
parent
[Synset('entity.n.01')]
26
two terms
parent
[Synset('abstraction.n.06')]
27
two terms
parent
[Synset('abstraction.n.06')]
28
two terms
parent
[Synset('abstraction.n.06')]
29
two terms
parent
[Synset('entity.n.01')]
30
two terms
parent
[Synset('entity.n.01')]
31
two terms
parent
[Synset('entity.n.01')]
3

[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]
152
two terms
parent
[Synset('entity.n.01')]
153
two terms
parent
[Synset('abstraction.n.06')]
154
two terms
parent
[Synset('entity.n.01')]
155
two terms
parent
[Synset('entity.n.01')]
156
two terms
parent
[Synset('abstraction.n.06')]
157
two terms
parent
[Synset('psychological_feature.n.01')]
1
two terms
parent
[Synset('object.n.01')]
1
two terms
parent
[Synset('object.n.01')]
3
two terms
parent
[Synset('whole.n.02')]
3
two terms
parent
[Synset('whole.n.02')]
4
two terms
parent
[Synset('entity.n.01')]
5
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
8
two terms
parent
[Synset('whole.n.02')]
9
two terms
parent
[Synset('physical_entity.n.01')]
10
two terms
parent
[Synset('entity.n.01')]
11
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
15
two terms
pare

128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('psychological_feature.n.01')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('psychological_feature.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
par

123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two ter

110
two terms
parent
[]
111
two terms
parent
[Synset('physical_entity.n.01')]
112
two terms
parent
[Synset('entity.n.01')]
113
two terms
parent
[Synset('physical_entity.n.01')]
114
two terms
parent
[Synset('object.n.01')]
115
two terms
parent
[Synset('entity.n.01')]
116
two terms
parent
[]
117
two terms
parent
[Synset('physical_entity.n.01')]
118
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('location.n.01')]
121
two terms
parent
[Synset('location.n.01')]
121
two terms
parent
[Synset('location.n.01')]
122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
1

116
two terms
parent
[]
117
two terms
parent
[Synset('abstraction.n.06')]
118
two terms
parent
[Synset('abstraction.n.06')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('attribute.n.02')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('attribute.n.02')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent

126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('abstraction.n.06')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two ter

128
two terms
parent
[Synset('matter.n.03')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('physical_entity.n.01')]
132
two terms
parent
[Synset('matter.n.03')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('matter.n.03')]
135
two terms
parent
[Synset('matter.n.03')]
136
two terms
parent
[Synset('physical_entity.n.01')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('physical_entity.n.01')]
139
two terms
parent
[Synset('physical_entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('entity.n.01')]
142
two terms
parent
[Synset('entity.n.01')]
143
two terms
parent
[Synset('matter.n.03')]
144
two terms
parent
[Synset('entity.n.01')]
145
two terms
parent
[Synset('entity.n.01')]
146
two terms
parent
[Synset('matter.n.03')]
147
two terms
parent
[]
148
two terms
parent
[Synset('entity.n.01')]
149
two terms
parent
[Synset('phys

[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('physical_entity.n.01')]
135
two terms
parent
[Synset('physical_entity.n.01')]
136
two terms
parent
[Synset('object.n.01')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('object.n.01')]
139
two terms
parent
[Synset('object.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('entity.n.01')]
142
two terms
parent
[Synset('entity.n.01')]
143
two terms
parent
[Synset('physical_entity.n.01')]
144
two terms
parent
[Synset('entity.n.01')]
145
two terms
parent
[Synset('entity.n.01')]
146
two terms
parent
[Synset('physical_entity.n.01')]
147
two terms
parent
[]
148
two terms
parent
[Synset('entity.n.01')]
149
two terms
parent
[Synset('physical_entity.n.01')]
150
two terms
parent
[Synset('entity.n.01')]
151
two terms
parent
[Synset('entity.n.01')]
152
two terms
parent
[Synset('instrumentality.n.03')]
153
two terms
parent
[Synset('entity.n.01')]
1

122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two ter

121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('whole.n.02')]
132
two terms
parent
[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('physical_entity.n.01')]
135
two terms
parent
[Synset('physical_entity.n.01')]
136
two terms
parent
[Synset('object.n.01')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('object.n.01')]
139
two ter

105
two terms
parent
[Synset('whole.n.02')]
106
two terms
parent
[Synset('physical_entity.n.01')]
107
two terms
parent
[Synset('entity.n.01')]
108
two terms
parent
[Synset('entity.n.01')]
109
two terms
parent
[Synset('physical_entity.n.01')]
110
two terms
parent
[]
111
two terms
parent
[Synset('physical_entity.n.01')]
112
two terms
parent
[Synset('entity.n.01')]
113
two terms
parent
[Synset('physical_entity.n.01')]
114
two terms
parent
[Synset('whole.n.02')]
115
two terms
parent
[Synset('entity.n.01')]
116
two terms
parent
[]
117
two terms
parent
[Synset('physical_entity.n.01')]
118
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physic

121
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
122
two terms
parent
[Synset('oil.n.01')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('substance.n.01')]
128
two terms
parent
[Synset('substance.n.01')]
128
two terms
parent
[Synset('substance.n.01')]
129
two terms
parent
[Synset('molecule.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('physical_entity.n.01')]
132
two terms
parent
[Synset('oil.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('oil.n.01')]
135
two terms
parent
[Synset('substance.n.01')]
136
two terms
parent
[Synset('physical_entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('physical_entity.n.01')]
139
two terms
parent
[Synset('physical_entity.n.01')]
140

68
two terms
parent
[]
68
two terms
parent
[]
69
two terms
parent
[Synset('entity.n.01')]
70
two terms
parent
[Synset('abstraction.n.06')]
71
two terms
parent
[Synset('entity.n.01')]
72
two terms
parent
[Synset('act.n.02')]
73
two terms
parent
[Synset('abstraction.n.06')]
74
two terms
parent
[Synset('entity.n.01')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
78
two terms
parent
[Synset('abstraction.n.06')]
79
two terms
parent
[Synset('entity.n.01')]
80
two terms
parent
[Synset('activity.n.01')]
81
two terms
parent
[Synset('entity.n.01')]
83
two terms
parent
[Synset('abstraction.n.06')]
83
two terms
parent
[Synset('abstraction.n.06')]
84
two terms
parent
[Synset('psychological_feature.n.01')]
85
two terms
parent
[Synset('event.n.01')]
86
two terms
parent
[Synset('abstraction.n.06')]
88
two terms
parent
[Synset('entity.n.01')]
88
two terms
parent
[Synset('entity.n.01')]
89
two terms
par

68
two terms
parent
[]
69
two terms
parent
[Synset('entity.n.01')]
70
two terms
parent
[Synset('abstraction.n.06')]
71
two terms
parent
[Synset('entity.n.01')]
72
two terms
parent
[Synset('action.n.01')]
73
two terms
parent
[Synset('abstraction.n.06')]
74
two terms
parent
[Synset('entity.n.01')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
78
two terms
parent
[Synset('abstraction.n.06')]
79
two terms
parent
[Synset('entity.n.01')]
80
two terms
parent
[Synset('act.n.02')]
81
two terms
parent
[Synset('entity.n.01')]
83
two terms
parent
[Synset('abstraction.n.06')]
83
two terms
parent
[Synset('abstraction.n.06')]
84
two terms
parent
[Synset('psychological_feature.n.01')]
85
two terms
parent
[Synset('event.n.01')]
86
two terms
parent
[Synset('abstraction.n.06')]
88
two terms
parent
[Synset('entity.n.01')]
88
two terms
parent
[Synset('entity.n.01')]
89
two terms
parent
[Synset('action.n.01'

34
two terms
parent
[Synset('entity.n.01')]
35
two terms
parent
[Synset('entity.n.01')]
36
two terms
parent
[Synset('entity.n.01')]
38
two terms
parent
[Synset('car.n.01')]
similar terms
Synset('car.n.01')
Synset('car.n.01')
38
two terms
parent
[Synset('car.n.01')]
similar terms
Synset('car.n.01')
Synset('car.n.01')
39
two terms
parent
[Synset('instrumentality.n.03')]
41
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('object.n.01')]
43
two terms
parent
[Synset('object.n.01')]
44
two terms
parent
[Synset('entity.n.01')]
45
two terms
parent
[Synset('entity.n.01')]
46
two terms
parent
[Synset('physical_entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('instrumentality.n.03')]
50
two terms
parent
[Synset('object.n.01')]
50
two terms
parent
[Synset('object.n.01')]
51
two terms
parent
[Synset('entity.n.01')]
52
two terms
parent
[Synset('instrumentality.n.03')]
53
two terms
parent
[Synset('physical_entity.n.01')

similar terms
Synset('engine.n.01')
Synset('engine.n.01')
153
two terms
parent
[Synset('entity.n.01')]
154
two terms
parent
[Synset('engine.n.01')]
similar terms
Synset('engine.n.01')
Synset('engine.n.01')
155
two terms
parent
[Synset('physical_entity.n.01')]
156
two terms
parent
[Synset('entity.n.01')]
157
two terms
parent
[Synset('entity.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
3
two terms
parent
[Synset('entity.n.01')]
3
two terms
parent
[Synset('entity.n.01')]
4
two terms
parent
[Synset('abstraction.n.06')]
5
two terms
parent
[Synset('abstraction.n.06')]
7
two terms
parent
[Synset('event.n.01')]
7
two terms
parent
[Synset('event.n.01')]
8
two terms
parent
[Synset('entity.n.01')]
9
two terms
parent
[Synset('abstraction.n.06')]
10
two terms
parent
[Synset('event.n.01')]
11
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
15
two terms
parent
[Synset('ent

147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
parent
[Synset('entity.n.01')]
150
two terms
parent
[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]
152
two terms
parent
[Synset('entity.n.01')]
153
two terms
parent
[Synset('abstraction.n.06')]
154
two terms
parent
[Synset('entity.n.01')]
155
two terms
parent
[Synset('entity.n.01')]
156
two terms
parent
[Synset('abstraction.n.06')]
157
two terms
parent
[Synset('event.n.01')]
1
two terms
parent
[Synset('object.n.01')]
1
two terms
parent
[Synset('object.n.01')]
3
two terms
parent
[Synset('artifact.n.01')]
3
two terms
parent
[Synset('artifact.n.01')]
4
two terms
parent
[Synset('entity.n.01')]
5
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
7
two terms
parent
[Synset('entity.n.01')]
8
two terms
parent
[Synset('engine.n.01')]
similar terms
Synset('engine.n.01')
Synset('engine.n.01')
9
two terms
parent
[Synset('physical_entity.n.01')]
1

132
two terms
parent
[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('physical_entity.n.01')]
135
two terms
parent
[Synset('physical_entity.n.01')]
136
two terms
parent
[Synset('region.n.03')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('location.n.01')]
139
two terms
parent
[Synset('location.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('entity.n.01')]
142
two terms
parent
[Synset('entity.n.01')]
143
two terms
parent
[Synset('physical_entity.n.01')]
144
two terms
parent
[Synset('entity.n.01')]
145
two terms
parent
[Synset('entity.n.01')]
146
two terms
parent
[Synset('physical_entity.n.01')]
147
two terms
parent
[]
148
two terms
parent
[Synset('entity.n.01')]
149
two terms
parent
[Synset('physical_entity.n.01')]
150
two terms
parent
[Synset('entity.n.01')]
151
two terms
parent
[Synset('entity.n.01')]
152
two terms
parent
[Synset('object.n.01')]
153
two terms
parent
[Synset('

139
two terms
parent
[Synset('object.n.01')]
139
two terms
parent
[Synset('object.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('event.n.01')]
143
two terms
parent
[Synset('abstraction.n.06'), Synset('physical_entity.n.01')]
144
two terms
parent
[Synset('event.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06'), Synset('physical_entity.n.01')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
parent
[Synset('physical_entity.n.01')]
150
two terms
parent
[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]
152
two terms
parent
[Synset('artifact.n.01')]
153
two terms
parent
[Synset('abstraction.n.06')]
154
two terms
parent
[Synset('artifact.n.01')]
155
two terms
parent
[Synset('physical_entity.n.01')]
156
two terms
parent
[Synset('abstraction.n.06')]
157
two terms
parent
[Synset('event.n.01')]
1
two terms
pa

123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('part.n.01')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('part.n.01')]
135
two terms
parent
[Synset('part.n.01')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two terms
parent
[Synset('part.n.01')]
144
two terms
parent
[Synset('abstraction.n.06')]
145
two te

105
two terms
parent
[Synset('entity.n.01')]
106
two terms
parent
[Synset('part.n.01')]
107
two terms
parent
[Synset('abstraction.n.06')]
108
two terms
parent
[Synset('abstraction.n.06')]
109
two terms
parent
[Synset('part.n.01')]
110
two terms
parent
[]
111
two terms
parent
[Synset('entity.n.01')]
112
two terms
parent
[Synset('abstraction.n.06')]
113
two terms
parent
[Synset('part.n.01')]
114
two terms
parent
[Synset('abstraction.n.06')]
115
two terms
parent
[Synset('abstraction.n.06')]
116
two terms
parent
[]
117
two terms
parent
[Synset('part.n.01')]
118
two terms
parent
[Synset('part.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('part.n.01')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Syns

95
two terms
parent
[Synset('entity.n.01')]
96
two terms
parent
[Synset('abstraction.n.06')]
98
two terms
parent
[Synset('abstraction.n.06')]
98
two terms
parent
[Synset('abstraction.n.06')]
99
two terms
parent
[Synset('abstraction.n.06')]
100
two terms
parent
[Synset('abstraction.n.06')]
102
two terms
parent
[Synset('entity.n.01')]
102
two terms
parent
[Synset('entity.n.01')]
103
two terms
parent
[Synset('entity.n.01')]
104
two terms
parent
[Synset('entity.n.01')]
105
two terms
parent
[Synset('entity.n.01')]
106
two terms
parent
[Synset('abstraction.n.06')]
107
two terms
parent
[Synset('abstraction.n.06')]
108
two terms
parent
[Synset('attribute.n.02')]
109
two terms
parent
[Synset('abstraction.n.06')]
110
two terms
parent
[]
111
two terms
parent
[Synset('entity.n.01')]
112
two terms
parent
[Synset('abstraction.n.06')]
113
two terms
parent
[Synset('abstraction.n.06')]
114
two terms
parent
[Synset('abstraction.n.06')]
115
two terms
parent
[Synset('attribute.n.02')]
116
two terms
parent

81
two terms
parent
[Synset('entity.n.01')]
83
two terms
parent
[Synset('abstraction.n.06')]
83
two terms
parent
[Synset('abstraction.n.06')]
84
two terms
parent
[Synset('abstraction.n.06')]
85
two terms
parent
[Synset('abstraction.n.06')]
86
two terms
parent
[Synset('abstraction.n.06')]
88
two terms
parent
[Synset('entity.n.01')]
88
two terms
parent
[Synset('entity.n.01')]
89
two terms
parent
[Synset('abstraction.n.06')]
90
two terms
parent
[Synset('abstraction.n.06')]
91
two terms
parent
[Synset('entity.n.01')]
92
two terms
parent
[Synset('entity.n.01')]
93
two terms
parent
[Synset('abstraction.n.06')]
95
two terms
parent
[Synset('entity.n.01')]
95
two terms
parent
[Synset('entity.n.01')]
96
two terms
parent
[Synset('abstraction.n.06')]
98
two terms
parent
[Synset('abstraction.n.06')]
98
two terms
parent
[Synset('abstraction.n.06')]
99
two terms
parent
[Synset('abstraction.n.06')]
100
two terms
parent
[Synset('abstraction.n.06')]
102
two terms
parent
[Synset('entity.n.01')]
102
two t

64
two terms
parent
[Synset('act.n.02')]
66
two terms
parent
[Synset('abstraction.n.06')]
66
two terms
parent
[Synset('abstraction.n.06')]
68
two terms
parent
[]
68
two terms
parent
[]
69
two terms
parent
[Synset('entity.n.01')]
70
two terms
parent
[Synset('abstraction.n.06')]
71
two terms
parent
[Synset('entity.n.01')]
72
two terms
parent
[Synset('act.n.02')]
73
two terms
parent
[Synset('abstraction.n.06')]
74
two terms
parent
[Synset('entity.n.01')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
77
two terms
parent
[Synset('abstraction.n.06')]
78
two terms
parent
[Synset('abstraction.n.06')]
79
two terms
parent
[Synset('entity.n.01')]
80
two terms
parent
[Synset('activity.n.01')]
81
two terms
parent
[Synset('entity.n.01')]
83
two terms
parent
[Synset('abstraction.n.06')]
83
two terms
parent
[Synset('abstraction.n.06')]
84
two terms
parent
[Synset('psychological_feature.n.01')]
85
two terms
parent
[Synset('event.n.01')]
86
two terms
p

48
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
51
two terms
parent
[Synset('condition.n.01')]
52
two terms
parent
[Synset('entity.n.01')]
53
two terms
parent
[Synset('entity.n.01')]
54
two terms
parent
[Synset('abstraction.n.06')]
56
two terms
parent
[Synset('abstraction.n.06')]
56
two terms
parent
[Synset('abstraction.n.06')]
57
two terms
parent
[Synset('attribute.n.02')]
58
two terms
parent
[Synset('abstraction.n.06')]
59
two terms
parent
[Synset('abstraction.n.06')]
61
two terms
parent
[Synset('abstraction.n.06')]
61
two terms
parent
[Synset('abstraction.n.06')]
62
two terms
parent
[Synset('attribute.n.02')]
63
two terms
parent
[Synset('abstraction.n.06')]
64
two terms
parent
[Synset('abstraction.n.06')]
66
two terms
parent
[Synset('abstraction.n.06')]
66
two terms
parent
[Synset('abstraction.n.06')]
68
two terms
parent
[]
68
two terms
parent
[]
69
two terms
parent
[Synset('entity.n.01')]
70
two ter

33
two terms
parent
[Synset('physical_entity.n.01')]
34
two terms
parent
[Synset('entity.n.01')]
35
two terms
parent
[Synset('entity.n.01')]
36
two terms
parent
[Synset('entity.n.01')]
38
two terms
parent
[Synset('whole.n.02')]
38
two terms
parent
[Synset('whole.n.02')]
39
two terms
parent
[Synset('whole.n.02')]
41
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('object.n.01')]
43
two terms
parent
[Synset('object.n.01')]
44
two terms
parent
[Synset('entity.n.01')]
45
two terms
parent
[Synset('entity.n.01')]
46
two terms
parent
[Synset('physical_entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('whole.n.02')]
50
two terms
parent
[Synset('object.n.01')]
50
two terms
parent
[Synset('object.n.01')]
51
two terms
parent
[Synset('entity.n.01')]
52
two terms
parent
[Synset('whole.n.02')]
53
two terms
parent
[Synset('physical_entity.n.01')]
54
two terms
parent
[Synset('physical_entity.n.01')]
56
two terms
parent
[S

31
two terms
parent
[Synset('entity.n.01')]
32
two terms
parent
[Synset('entity.n.01')]
33
two terms
parent
[Synset('entity.n.01')]
34
two terms
parent
[Synset('attribute.n.02')]
35
two terms
parent
[Synset('attribute.n.02')]
36
two terms
parent
[Synset('abstraction.n.06')]
38
two terms
parent
[Synset('entity.n.01')]
38
two terms
parent
[Synset('entity.n.01')]
39
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('abstraction.n.06')]
41
two terms
parent
[Synset('abstraction.n.06')]
43
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('entity.n.01')]
44
two terms
parent
[Synset('attribute.n.02')]
45
two terms
parent
[Synset('abstraction.n.06')]
46
two terms
parent
[Synset('entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
50
two terms
parent
[Synset('entity.n.01')]
51
two terms
parent
[Synset('condition.n.01')]
52
two terms
parent
[Synset('entity.n.01')]
53
two terms
parent
[

10
two terms
parent
[Synset('entity.n.01')]
11
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
13
two terms
parent
[Synset('whole.n.02')]
15
two terms
parent
[Synset('object.n.01')]
15
two terms
parent
[Synset('object.n.01')]
16
two terms
parent
[Synset('physical_entity.n.01')]
18
two terms
parent
[Synset('physical_entity.n.01')]
18
two terms
parent
[Synset('physical_entity.n.01')]
19
two terms
parent
[Synset('entity.n.01')]
20
two terms
parent
[Synset('physical_entity.n.01')]
21
two terms
parent
[Synset('whole.n.02')]
22
two terms
parent
[Synset('entity.n.01')]
23
two terms
parent
[Synset('physical_entity.n.01')]
24
two terms
parent
[Synset('physical_entity.n.01')]
25
two terms
parent
[Synset('whole.n.02')]
26
two terms
parent
[Synset('entity.n.01')]
27
two terms
parent
[Synset('entity.n.01')]
28
two terms
parent
[Synset('entity.n.01')]
29
two terms
parent
[Synset('whole.n.02')]
30
two terms
parent
[Synset('physical_entity.n.01')]
31
two terms
parent

139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('psychological_feature.n.01')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('psychological_feature.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
parent
[Synset('entity.n.01')]
150
two terms
parent
[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]
152
two terms
parent
[Synset('entity.n.01')]
153
two terms
parent
[Synset('abstraction.n.06')]
154
two terms
parent
[Synset('entity.n.01')]
155
two terms
parent
[Synset('entity.n.01')]
156
two terms
parent
[Synset('abstraction.n.06')]
157
two terms
parent
[Synset('psychological_feature.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
3
two terms

128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('psychological_feature.n.01')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('psychological_feature.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
par

38
two terms
parent
[Synset('whole.n.02')]
39
two terms
parent
[Synset('whole.n.02')]
41
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('object.n.01')]
43
two terms
parent
[Synset('object.n.01')]
44
two terms
parent
[Synset('entity.n.01')]
45
two terms
parent
[Synset('entity.n.01')]
46
two terms
parent
[Synset('physical_entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('whole.n.02')]
50
two terms
parent
[Synset('object.n.01')]
50
two terms
parent
[Synset('object.n.01')]
51
two terms
parent
[Synset('entity.n.01')]
52
two terms
parent
[Synset('whole.n.02')]
53
two terms
parent
[Synset('physical_entity.n.01')]
54
two terms
parent
[Synset('physical_entity.n.01')]
56
two terms
parent
[Synset('physical_entity.n.01')]
56
two terms
parent
[Synset('physical_entity.n.01')]
57
two terms
parent
[Synset('entity.n.01')]
58
two terms
parent
[Synset('entity.n.01')]
59
two terms
parent
[Synset('physical_entity.n.01')]
61


35
two terms
parent
[Synset('entity.n.01')]
36
two terms
parent
[Synset('entity.n.01')]
38
two terms
parent
[Synset('physical_entity.n.01')]
38
two terms
parent
[Synset('physical_entity.n.01')]
39
two terms
parent
[Synset('physical_entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('physical_entity.n.01')]
43
two terms
parent
[Synset('physical_entity.n.01')]
44
two terms
parent
[Synset('entity.n.01')]
45
two terms
parent
[Synset('entity.n.01')]
46
two terms
parent
[Synset('physical_entity.n.01')]
47
two terms
parent
[]
48
two terms
parent
[Synset('physical_entity.n.01')]
50
two terms
parent
[Synset('physical_entity.n.01')]
50
two terms
parent
[Synset('physical_entity.n.01')]
51
two terms
parent
[Synset('entity.n.01')]
52
two terms
parent
[Synset('physical_entity.n.01')]
53
two terms
parent
[Synset('physical_entity.n.01')]
54
two terms
parent
[Synset('matter.n.03')]
56
two terms
parent
[Synset('matter.n.03')

25
two terms
parent
[Synset('object.n.01')]
26
two terms
parent
[Synset('entity.n.01')]
27
two terms
parent
[Synset('entity.n.01')]
28
two terms
parent
[Synset('entity.n.01')]
29
two terms
parent
[Synset('object.n.01')]
30
two terms
parent
[Synset('physical_entity.n.01')]
31
two terms
parent
[Synset('area.n.01')]
32
two terms
parent
[Synset('object.n.01')]
33
two terms
parent
[Synset('physical_entity.n.01')]
34
two terms
parent
[Synset('entity.n.01')]
35
two terms
parent
[Synset('entity.n.01')]
36
two terms
parent
[Synset('entity.n.01')]
38
two terms
parent
[Synset('object.n.01')]
38
two terms
parent
[Synset('object.n.01')]
39
two terms
parent
[Synset('object.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
41
two terms
parent
[Synset('entity.n.01')]
43
two terms
parent
[Synset('location.n.01')]
43
two terms
parent
[Synset('location.n.01')]
44
two terms
parent
[Synset('entity.n.01')]
45
two terms
parent
[Synset('entity.n.01')]
46
two terms
parent
[Synset('physical_entity.n.01')]
47


154
two terms
parent
[Synset('object.n.01')]
155
two terms
parent
[Synset('physical_entity.n.01')]
156
two terms
parent
[Synset('entity.n.01')]
157
two terms
parent
[Synset('entity.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
1
two terms
parent
[Synset('entity.n.01')]
3
two terms
parent
[Synset('entity.n.01')]
3
two terms
parent
[Synset('entity.n.01')]
4
two terms
parent
[Synset('attribute.n.02')]
5
two terms
parent
[Synset('abstraction.n.06')]
7
two terms
parent
[Synset('abstraction.n.06')]
7
two terms
parent
[Synset('abstraction.n.06')]
8
two terms
parent
[Synset('entity.n.01')]
9
two terms
parent
[Synset('abstraction.n.06')]
10
two terms
parent
[Synset('abstraction.n.06')]
11
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
13
two terms
parent
[Synset('entity.n.01')]
15
two terms
parent
[Synset('entity.n.01')]
15
two terms
parent
[Synset('entity.n.01')]
16
two terms
parent
[Synset('abstraction.n.06')]
18
two terms
parent
[Synset('abstraction

124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('device.n.01')]
132
two terms
parent
[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('physical_entity.n.01')]
135
two terms
parent
[Synset('physical_entity.n.01')]
136
two terms
parent
[Synset('object.n.01')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('object.n.01')]
139
two terms
parent
[Synset('object.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('entity.n.01')]
142
two terms
parent
[Synset('entity.n.01')]
143
two terms
parent
[Synset('physical_entity.n.01')]
144
two terms
parent
[Synset('e

121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('measure.n.02')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140


134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('psychological_feature.n.01')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('psychological_feature.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
parent
[Synset('entity.n.01')]
150
two terms
parent
[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]
152
two terms
parent
[Synset('entity.n.01')]
153
two terms
parent
[Synset('abstraction.n.06')]
154
two terms
parent
[Synset('entity.n.01')]
155
two term

111
two terms
parent
[Synset('physical_entity.n.01')]
112
two terms
parent
[Synset('abstraction.n.06')]
113
two terms
parent
[Synset('oil.n.01')]
similar terms
Synset('oil.n.01')
Synset('oil.n.01')
114
two terms
parent
[Synset('abstraction.n.06'), Synset('physical_entity.n.01')]
115
two terms
parent
[Synset('abstraction.n.06')]
116
two terms
parent
[]
117
two terms
parent
[Synset('oil.n.01')]
similar terms
Synset('oil.n.01')
Synset('oil.n.01')
118
two terms
parent
[Synset('substance.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
122
two terms
parent
[Synset('oil.n.01')]
similar terms
Synset('oil.n.01')
Synset('oil.n.01')
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('substance.n.01')]
128
two terms
parent
[Synset('substance.n.01')]
12

110
two terms
parent
[]
111
two terms
parent
[Synset('entity.n.01')]
112
two terms
parent
[Synset('psychological_feature.n.01')]
113
two terms
parent
[Synset('abstraction.n.06')]
114
two terms
parent
[Synset('psychological_feature.n.01')]
115
two terms
parent
[Synset('abstraction.n.06')]
116
two terms
parent
[]
117
two terms
parent
[Synset('abstraction.n.06')]
118
two terms
parent
[Synset('abstraction.n.06')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two term

111
two terms
parent
[Synset('physical_entity.n.01')]
112
two terms
parent
[Synset('entity.n.01')]
113
two terms
parent
[Synset('physical_entity.n.01')]
114
two terms
parent
[Synset('whole.n.02')]
115
two terms
parent
[Synset('entity.n.01')]
116
two terms
parent
[]
117
two terms
parent
[Synset('physical_entity.n.01')]
118
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('wh

121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]


124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('part.n.01')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('part.n.01')]
135
two terms
parent
[Synset('part.n.01')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two terms
parent
[Synset('part.n.01')]
144
two terms
parent
[Synset('abstraction.n.06')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two te

122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('artifact.n.01')]
132
two terms
parent
[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('physical_entity.n.01')]
135
two terms
parent
[Synset('physical_entity.n.01')]
136
two terms
parent
[Synset('object.n.01')]
137
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('object.n.01')]
139
two terms
parent
[Synset('object.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('entity.n.01')]
142
two terms
parent
[Synset(

130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
parent
[Synset('event.n.01')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('event.n.01')]
145
two terms
parent
[Synset('abstraction.n.06')]
146
two terms
parent
[Synset('abstraction.n.06')]
147
two terms
parent
[]
148
two terms
parent
[Synset('abstraction.n.06')]
149
two terms
parent
[Synset('entity.n.01')]
150
two terms
parent
[Synset('abstraction.n.06')]
151
two terms
parent
[Synset('abstraction.n.06')]

117
two terms
parent
[Synset('oil.n.01')]
118
two terms
parent
[Synset('substance.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('physical_entity.n.01')]
122
two terms
parent
[Synset('oil.n.01')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('substance.n.01')]
128
two terms
parent
[Synset('substance.n.01')]
128
two terms
parent
[Synset('substance.n.01')]
129
two terms
parent
[Synset('molecule.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('physical_entity.n.01')]
132
two terms
parent
[Synset('oil.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('oil.n.01')]
135
two terms
parent
[Synset('substance.n.01')]
136
two terms
parent
[Synset('physical_entity.n.01')]
137
two terms
par

114
two terms
parent
[Synset('abstraction.n.06')]
115
two terms
parent
[Synset('abstraction.n.06')]
116
two terms
parent
[]
117
two terms
parent
[Synset('part.n.01')]
118
two terms
parent
[Synset('part.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('part.n.01')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('part.n.01')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('part.n.01')]
135
two terms
parent
[Synset('part.n.01')]
136
two terms
parent
[Sy

115
two terms
parent
[Synset('abstraction.n.06')]
116
two terms
parent
[]
117
two terms
parent
[Synset('part.n.01')]
118
two terms
parent
[Synset('part.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('part.n.01')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
128
two terms
parent
[Synset('part.n.01')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('part.n.01')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('part.n.01')]
135
two terms
parent
[Synset('part.n.01')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset(

124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('attribute.n.02')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('attribute.n.02')]
142
two terms
parent
[Synset('abstraction.n.06')]
143
two terms
parent
[Synset('abstraction.n.06')]
144
two terms
parent
[Synset('abstraction.n.06')]
145
two terms
p

121
two terms
parent
[Synset('entity.n.01')]
122
two terms
parent
[Synset('abstraction.n.06')]
123
two terms
parent
[Synset('abstraction.n.06')]
124
two terms
parent
[Synset('abstraction.n.06')]
125
two terms
parent
[Synset('abstraction.n.06')]
126
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
128
two terms
parent
[Synset('abstraction.n.06')]
129
two terms
parent
[Synset('entity.n.01')]
130
two terms
parent
[Synset('entity.n.01')]
131
two terms
parent
[Synset('entity.n.01')]
132
two terms
parent
[Synset('abstraction.n.06')]
133
two terms
parent
[Synset('entity.n.01')]
134
two terms
parent
[Synset('abstraction.n.06')]
135
two terms
parent
[Synset('abstraction.n.06')]
136
two terms
parent
[Synset('entity.n.01')]
137
two terms
parent
[Synset('abstraction.n.06')]
139
two terms
parent
[Synset('entity.n.01')]
139
two terms
parent
[Synset('entity.n.01')]
140
two terms
parent
[]
141
two terms
parent
[Synset('abstraction.n.06')]
142
two terms
pa

114
two terms
parent
[Synset('whole.n.02')]
115
two terms
parent
[Synset('entity.n.01')]
116
two terms
parent
[]
117
two terms
parent
[Synset('physical_entity.n.01')]
118
two terms
parent
[Synset('physical_entity.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
121
two terms
parent
[Synset('object.n.01')]
122
two terms
parent
[Synset('physical_entity.n.01')]
123
two terms
parent
[Synset('entity.n.01')]
124
two terms
parent
[Synset('entity.n.01')]
125
two terms
parent
[Synset('entity.n.01')]
126
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
128
two terms
parent
[Synset('physical_entity.n.01')]
129
two terms
parent
[Synset('physical_entity.n.01')]
130
two terms
parent
[Synset('physical_entity.n.01')]
131
two terms
parent
[Synset('whole.n.02')]
132
two terms
parent
[Synset('physical_entity.n.01')]
133
two terms
parent
[Synset('physical_entity.n.01')]
134
two terms
parent
[Synset('phy